tests/test_enterprise/api/test_views.py

```diff
@@ -2385,6 +2385,77 @@ def test_post_licensed_course_enrollments_license_revoke(
         assert enterprise_course_enrollment.saved_for_later == is_revoked
         assert licensed_course_enrollment.is_revoked == is_revoked
 
+    @ddt.data(
+        # API should return enterprise customer details
+        (False, True, True),
+        # API should not return enterprise customer details
+        (False, False, False),
+        # for staff, API should return enterprise customer details irrespective of linked status
+        (True, True, True),
+        # for staff, API should return enterprise customer details irrespective of linked status
+        (True, False, True),
+    )
+    @ddt.unpack
+    def test_enterprise_customer_filter(self, is_staff, is_linked_to_enterprise, has_access):
+        enterprise_customer_data = {
+            'uuid': FAKE_UUIDS[0], 'name': 'Test Enterprise Customer', 'slug': TEST_SLUG,
+            'active': True, 'enable_data_sharing_consent': True,
+            'enforce_data_sharing_consent': 'at_enrollment', 'enable_portal_code_management_screen': True,
+            'enable_portal_reporting_config_screen': False,
+            'enable_portal_saml_configuration_screen': False,
+            'site__domain': 'example.com', 'site__name': 'example.com',
+            'enable_portal_subscription_management_screen': False,
+            'contact_email': 'fake@example.com',
+        }
+        enterprise_customer = factories.EnterpriseCustomerFactory(**enterprise_customer_data)
+
+        user = factories.UserFactory(username='test_user', is_active=True, is_staff=is_staff)
```

Comment:
Seems like another thing that might be good as a utility function?

Reply:
One small step could be making the username and password here constants in the test class data so you can just use the variable